Onboarding migration (#340)
apify/apify-docs: content/academy/pro_apify_scraper_development/apify_api_and_client.md

---
title: Apify API & client
description: Gain an in-depth understanding of the two main ways of programmatically interacting with the Apify platform - throught the API, and through a client.
menuWeight: 6.5
paths:
    - pro-apify-scraper-development/apify-api-and-client
---

# [](#api-and-client) Apify API & client

> There are two main ways to programmatically interact with the Apify platform: By directly using [Apify's RESTful API](https://docs.apify.com/api/v2), or by using the [JavaScript](https://docs.apify.com/apify-client-js) and [Python](https://docs.apify.com/apify-client-python) API clients. In the next two lessons, we'll be focusing on the first two.

Reviewer: ```suggestion
There are two main ways to programmatically interact with the Apify platform: by directly using [Apify's RESTful API](https://docs.apify.com/api/v2), or by using the [JavaScript](https://docs.apify.com/apify-client-js) and [Python](https://docs.apify.com/apify-client-python) API clients. In the next two lessons, we'll be focusing on the first two.
```

> > Apify's API and JavaScript API client allow us to do anything a regular user can do when interacting with the platform's web interfacee, but programmatically instead.

Reviewer: ```suggestion
> Apify's API and JavaScript API client allow us to do anything a regular user can do when interacting with the platform's web interface, only programmatically.
```

## Our task

In the previous lesson, we created a **task** for the Amazon actor we build in the first two lessons of this course. Now, we'll be creating another new actor, which will have two jobs:

1. Programmatically call the Amazon actor.
2. Export its results into CSV format under a new key called **OUTPUT.csv** in the default key-value store.

Though it's a bit unintuitive, this is a perfect activity for learning how to use both the Apify API and the Apify JavaScript client.

The new actor should take the following input values, which be mapped to parameters in the API calls:

```JSON
{
    // How much memory to allocate to the Amazon actor
    // Must be a power of 2
    "memory": 4096,

    // Whether to use the JavaScript client to make the
    // call, or to use the API
    "useClient": false,

    // The fields in each item to return back. All other
    // fields should be ommitted
    "fields": ["title", "itemUrl", "offer"],

    // The maximum number of items to return back
    "maxItems": 10
}
```

[**Solution**]({{@link pro_apify_scraper_development/solutions/using_api_and_client.md}})

## [](#next) Next up

> Wow, you've learned a whole lot so far. The next lesson will surely be released soon, as we release new lessons as we write them, so stay tuned!

Reviewer: Maybe rephrase to 
```suggestion
Wow, you've learned a whole lot so far. The next lesson will be released as soon as it's written, so stay tuned!
```
